sale_loyalty_limit/models/coupon_rule.py

```diff
@@ -19,42 +19,70 @@ def _compute_rule_salesmen_limit_count(self):
         return res
 
 
-class CouponRuleSalesmenLimit(models.Model):
-    _inherit = "coupon.rule.salesmen.limit"
+class LoyaltyRuleSalesmenLimit(models.Model):
+    _inherit = "loyalty.rule.salesmen.limit"
 
     def _compute_rule_times_used(self):
         """This count is also used in the check methods to avoid applying the rule
         above the salesmen limits."""
         res = super()._compute_rule_times_used()
-        programs = self.env["coupon.program"].search_read(
-            [("rule_id", "in", self.mapped("rule_id").ids)],
-            ["id", "rule_id", "program_type", "coupon_ids", "rule_salesmen_limit_ids"],
+        programs = self.env["loyalty.program"].search_read(
+            [("rule_ids", "in", self.mapped("rule_id").ids)],
+            ["id", "rule_ids", "program_type", "coupon_ids"],
         )
         coupon_programs = [
             p
             for p in programs
-            if p["program_type"]
-            and p["program_type"] == "coupon_program"
-            or p["coupon_ids"]
+            if p["program_type"] and p["program_type"] == "coupons" or p["coupon_ids"]
         ]
         for program in programs:
+            rule_salesmen_limit_ids = [
+                rule_limit.id
+                for rule_id in program["rule_ids"]
+                for rule_limit in self.env["loyalty.rule"]
+                .browse(rule_id)
+                .mapped("rule_salesmen_limit_ids")
+            ]
             salesmen_limits = self.filtered(
-                lambda x: x._origin.id in program["rule_salesmen_limit_ids"]
+                lambda x: x._origin.id in rule_salesmen_limit_ids
             )
             for salesman_limit in salesmen_limits:
                 if program in coupon_programs and not program["coupon_ids"]:
                     continue
                 elif program in coupon_programs:
+                    required_points = (
+                        self.env["loyalty.program"]
+                        .browse(program["id"])
+                        .reward_ids.mapped("required_points")
+                    )
+                    min_required_points = min(required_points) if required_points else 1
```

Comment:
Why do we need to check this?

Reply:
It's a mistake of mine that I didn't correct, defining a promotion without rewards gave me a failure in the tests but it doesn't make sense for a promotion to have no rewards.